src/bp/ui-studio/src/web/components/ContentForm/Text.tsx

```diff
@@ -12,21 +14,28 @@ interface Props {
 
 const Text: FC<Props> = props => {
   const [value, setValue] = useState('')
+  const { formContext, formData, schema, required, uiSchema, onChange } = props
+  const key = useRef(`${formContext?.customKey}`)
 
   useEffect(() => {
-    setValue(props.formData)
-  }, [props.formData])
+    setValue(formData)
+  }, [formData])
+
+  useEffect(() => {
```

Comment:
isn't there a way to implement a custom hook instead of re-implementing this in every content form components ?

Reply:
It's only implemented in two places, don't think there's enough value added to it, besides we might want to add a custom behaviour in the future when the key changes since it means that the form has changed

Reply:
It's only implemented in two places, don't think there's enough value added to it, besides we might want to add a custom behaviour in the future when the key changes since it means that the form has changed